lib/job-iteration/iteration.rb

```diff
@@ -288,6 +286,25 @@ def handle_completed(completed)
       raise "Unexpected thrown value: #{completed.inspect}"
     end
 
+    def cursor_position_from_job_data(job_data)
+      if job_data.key?("serialized_cursor_position")
+        deserialize_cursor(job_data.fetch("serialized_cursor_position"))
```

Comment:
I've updated this so if a serialized cursor is found, it should always be deserializable.